app/models/authenticator.rb

```diff
@@ -34,9 +34,23 @@ def uses_stored_password?
       false
     end
 
+    def can_authorize_user_by_userid?
+      false
+    end
+
+    def authorize_user_by_userid(userid)
+      return unless can_authorize_user_by_userid?
+      options = {
+        :require_user   => true,
+        :authorize_only => true
+      }
```

Comment:
Since you don't use these option elsewhere can you pass them directly to the next call.  It reads more explicit to me

```ruby
authenticate(userid, "", {},
  :require_user   => true,
  :authorize_only => true
)
```